sympy/physics/quantum/tensorproduct.py

```diff
@@ -345,6 +345,25 @@ def tensor_product_simp_Mul(e):
         return e
 
 
+def tensor_product_simp_Pow(e):
+    if not isinstance(e, Pow):
+        return e
+
+    new_arg = []
+    new_arg = tuple([arg**e.exp for arg in e.base.args])
+
+    if isinstance(e.base, Add):
+        return Add(*[tensor_product_simp(arg) for arg in new_arg])
```

Comment:
This seems to return the sum of powers of terms which is different from the power of the sum.
```
>>> p = TensorProduct(A, B)
>>> q = TensorProduct(C, D)
>>> tensor_product_simp_Pow((p + q)**2)
(A**2)x(B**2) + (C**2)x(D**2)
>>> tensor_product_simp(((p + q)*(p + q)).expand())
(A**2)x(B**2) + (C**2)x(D**2) + (A*C)x(B*D) + (C*A)x(D*B)
```

Reply:
@jksuom ,
I think if you do `tensor_product_simp_Pow(((p + q)**2).expand())` , you will get the result as `(A**2)x(B**2) + (C**2)x(D**2) + (A*C)x(B*D) + (C*A)x(D*B)` . Is this happening due to expand() ?

Reply:
```
>>> p = TensorProduct(A, B)
>>> q = TensorProduct(C, D)
>>> ((p + q)**2).expand()
AxB*CxD + AxB**2 + CxD*AxB + CxD**2
>>> tensor_product_simp_Pow(((p + q)**2).expand())
AxB*CxD + AxB**2 + CxD*AxB + CxD**2
```
`tensor_product_simp_Pow` has no effect on the expanded expression which is not a `Pow`.

Reply:
I cannot really understand what you're pointing at. Could you point me the code changes which I need to inflict? @jksuom

Reply:
The code on this line (L356) does not give the correct power of a sum since the cross terms like `(A*C)x(B*D)` and `(C*A)x(D*B)` are missing. With higher powers there should be even more cross terms.

Reply:
I am not understanding. I think that was the issue, that we have to first calculate the powers of the terms and then do whatever operation is necessary.
Could you please tell an alternative code to this, because I can't exactly understand the issue.
Thanks .. @jksuom

Reply:
I would not try to handle Pow objects whose base is an Add. I don't think there is any reasonably simple way of doing that with arbitrary exponents.

Reply:
So, should I only keep the MUL part for the POW, and skip the ADD, Commutator, and Anti-Commutator parts?
@jksuom

Reply:
I think that not much more than `Mul` can be implemented.

Reply:
@jksuom , I have removed the parts other than Mul, and its corresponding test cases too.
Please review.